mjml/core/registry.py

```diff
@@ -1,29 +1,64 @@
 
 
-__all__ = []
+__all__ = ['assign_components', 'components', 'handle_mjml_config_components', 'preset_core_components', 'register_component']
 
-def _components():
+import inspect
+from types import ModuleType
+from typing import Type
+
+components = {}
+
+def preset_core_components():
     from ..elements import (MjButton, MjText, MjSection, MjColumn, MjBody,
         MjGroup, MjImage, MjDivider, MjTable, MjRaw)
     from ..elements.head import (MjAttributes, MjFont, MjHead, MjPreview, MjStyle,
         MjTitle)
-    components = {
-        'mj-button': MjButton,
-        'mj-text': MjText,
-        'mj-divider': MjDivider,
-        'mj-image': MjImage,
-        'mj-section': MjSection,
-        'mj-column': MjColumn,
-        'mj-body': MjBody,
-        'mj-group'  : MjGroup,
-        'mj-table'  : MjTable,
-        'mj-raw'  : MjRaw,
+
+    assign_components(components, [
+        MjButton,
+        MjText,
+        MjDivider,
+        MjImage,
+        MjSection,
+        MjColumn,
+        MjBody,
+        MjGroup,
+        MjTable,
+        MjRaw,
         # --- head components ---
-        'mj-attributes': MjAttributes,
-        'mj-font': MjFont,
-        'mj-head': MjHead,
-        'mj-preview': MjPreview,
-        'mj-title': MjTitle,
-        'mj-style': MjStyle,
-    }
+        MjAttributes,
+        MjFont,
+        MjHead,
+        MjPreview,
+        MjTitle,
+        MjStyle,
+    ])
+
     return components
+
+def assign_components(target, source):
+    for component in source:
+        target[component.component_name] = component
+
+
+def register_component(Component: Type):
+    assign_components(components, [Component])
+
+
+def register_custom_component(
+    comp,
+    registerCompFn=register_component,
+):
+    try:
+        registerCompFn(comp)
+    except AttributeError:
+        if isinstance(comp, ModuleType):
+            # Convert module into list of submodules
+            comp = list([cls for name, cls in inspect.getmembers(comp) if inspect.isclass(cls)])
+
+        for component in comp:
+            register_custom_component(component, registerCompFn)
+
+
+def handle_mjml_config_components(components):
```

Comment:
I originally added this for parity with MJML, but it might make more sense to just rename `register_custom_component` to `register_custom_components` and use that instead.